### ISSUE TITLE
Web Content Accessibility Compliance

### ISSUE THREAD

**Issue by norinespears** [specify/specify7#772]
Request from members: 

The University of New England is evaluating the Specify application to replace our existing herbarium data management software and would appreciate if you could advise if version 7 is compliant to WCAG 2.0 or 2.1. 

Thanking you in advance,
Best regards,
Sylvia

**Comment by beach53** [comment]
Please indicate in potential member requests like this which collections and institutions the petitioner represents.  Thanks!

**Comment by beach53** [comment]
test

**Comment by maxxxxxdlp** [comment]
I began going over WCAG 2.1 requirements and implementing them.
It is available on the branch `a11y`.

**Comment by maxxxxxdlp** [comment]
Specify 7 Accessibility Statement - https://docs.google.com/document/d/1KXoNTUFQ1_MmyaJ_VRNTYXeyJc_lhcVCKiiVOOQ4x2o/edit#
An up-to-date list of accessibility issues - https://github.com/specify/specify7/projects/12

**Comment by maxxxxxdlp** [comment]
Need to go over all of these and check that Specify is compliant:
- a11yproject:
   - [ ] https://www.a11yproject.com/checklist/#identify-links-that-open-in-a-new-tab-or-window
   - [ ] https://www.a11yproject.com/checklist/#use-fieldset-and-legend-elements-where-appropriate
   - [ ] https://www.a11yproject.com/checklist/#make-sure-that-error-warning-and-success-states-are-not-visually-communicated-by-just-color
   - [ ] https://www.a11yproject.com/checklist/#check-your-content-in-specialized-browsing-modes
   - [ ] https://www.a11yproject.com/checklist/#increase-text-size-to-200percent
   - [ ] https://www.a11yproject.com/checklist/#double-check-that-good-proximity-between-content-is-maintained
   - [ ] https://www.a11yproject.com/checklist/#make-sure-color-isnt-the-only-way-information-is-conveyed
   - [ ] https://www.a11yproject.com/checklist/#ensure-animations-are-subtle-and-do-not-flash-too-much
   - [ ] https://www.a11yproject.com/checklist/#make-sure-all-animation-obeys-the-prefers-reduced-motion-media-query
   - [ ] https://www.a11yproject.com/checklist/#check-the-contrast-for-all-normal-sized-text
   - [ ] https://www.a11yproject.com/checklist/#check-the-contrast-for-all-large-sized-text
   - [ ] https://www.a11yproject.com/checklist/#check-the-contrast-for-all-icons
   - [ ] https://www.a11yproject.com/checklist/#check-the-contrast-of-borders-for-input-elements-text-input-radio-buttons-checkboxes-etc
   - [ ] https://www.a11yproject.com/checklist/#check-custom-selection-colors
   - [ ] https://www.a11yproject.com/checklist/#ensure-sufficient-space-between-interactive-items-in-order-to-provide-a-scroll-area

**Comment by maxxxxxdlp** [comment]
> Specify's current design is not compatible with having text appear above the object it is referring to

Do you mean the tooltips?
Many UI elements across specify have tooltips that activate when mouse is over the element.
On touchscreen devices, tooltips are visible on click (assuming the element is not a button/link).
For screen readers, the content of a tooltip is announced as a description when element is focused.

> Validation of data only communicates with the color green.

Validation results and error messages are exposed to the screen reader.
Though, I also plan to add the ability to customize the validation color.

> The contrast for these elements is 2.07:1 (#F1F2F2 and #AAAAAA are the colors for the text box outline and background). This does not meet a11y standards.

Thanks for checking that! I would go over contrast issues soon, as part of the global UI refactor.

**Comment by maxxxxxdlp** [comment]
The remaining checkboxes are all regarding the use of color. Factoring them out into a narrower-scoped issue - #3114
